cmd/scaffold.go

```diff
@@ -272,29 +281,70 @@ func (fnb *FlowNodeBuilder) InitFlowNetworkWithConduitFactory(node *NodeConfig,
 		myAddr = fnb.BaseConfig.BindAddr
 	}
 
+	var mwOpts []p2p.MiddlewareOption
+	if len(fnb.MsgValidators) > 0 {
+		mwOpts = append(mwOpts, p2p.WithMessageValidators(fnb.MsgValidators...))
+	}
+
+	idProviderPeerFilter := func(pid peer.ID) bool {
+		_, found := fnb.IdentityProvider.ByPeerID(pid)
+		return found
+	}
+
+	connGaterPeerDialFilters := p2p.PeerFilters{idProviderPeerFilter}
+	connGaterInterceptSecureFilters := p2p.PeerFilters{idProviderPeerFilter}
+
+	peerManagerFilters := p2p.PeerFilters{}
+
+	// setup unicast stream rate limiter
+	if fnb.BaseConfig.UnicastStreamCreationRateLimit > 0 && fnb.BaseConfig.UnicastStreamCreationBurstLimit > 0 {
+		unicastStreamsRateLimiter := unicast.NewStreamsRateLimiter(rate.Limit(fnb.BaseConfig.UnicastStreamCreationRateLimit), fnb.BaseConfig.UnicastStreamCreationBurstLimit)
+		mwOpts = append(mwOpts, p2p.WithUnicastStreamRateLimiter(unicastStreamsRateLimiter))
+
+		if !fnb.BaseConfig.UnicastRateLimitDryRun {
+			// add IsRateLimited peerFilters to conn gater intercept secure peer and peer manager filters list
+			connGaterInterceptSecureFilters = append(connGaterInterceptSecureFilters, unicastStreamsRateLimiter.IsRateLimited)
+			peerManagerFilters = append(peerManagerFilters, unicastStreamsRateLimiter.IsRateLimited)
+		}
+	}
+
+	// setup unicast bandwidth rate limiter
+	if fnb.BaseConfig.UnicastBandwidthRateLimit > 0 && fnb.BaseConfig.UnicastBandwidthBurstLimit > 0 {
+		unicastBandwidthRateLimiter := unicast.NewBandWidthRateLimiter(rate.Limit(fnb.BaseConfig.UnicastBandwidthRateLimit), fnb.BaseConfig.UnicastBandwidthBurstLimit)
+		mwOpts = append(mwOpts, p2p.WithUnicastStreamRateLimiter(unicastBandwidthRateLimiter))
+
+		if !fnb.BaseConfig.UnicastRateLimitDryRun {
+			// add IsRateLimited peerFilters to conn gater intercept secure peer and peer manager filters list
+			connGaterInterceptSecureFilters = append(connGaterInterceptSecureFilters, unicastBandwidthRateLimiter.IsRateLimited)
+			peerManagerFilters = append(peerManagerFilters, unicastBandwidthRateLimiter.IsRateLimited)
+		}
```

Comment:
to make sure I understand how this works, if a peer gets rate limited, then we will disconnect from them on the next peer manager update, and not immediately, right?